cishouseholds/pipeline/high_level_transformations.py

```diff
@@ -195,9 +196,24 @@ def transform_cis_soc_data(df: DataFrame, join_on_columns: List[str]) -> DataFra
     ).orderBy(F.desc("LENGTH"))
 
     window = Window.partitionBy(*join_on_columns)
-    df = df.withColumn("DROP", F.col("LENGTH") != F.max("LENGTH").over(window))
-    df = df.filter((F.col("standard_occupational_classification_code") != "uncodeable") & (~F.col("DROP")))
-    return df.drop("DROP", "LENGTH")
+
+    df = df.withColumn(
+        "DROP_REASON",
+        F.when(F.col("LENGTH") != F.max("LENGTH").over(window), "NOT MOST SPECIFIC").when(
```

Comment:
We only want to drop an uncodeable record if there is a record with a code and the same join keys. Otherwise, we want to keep the uncodeable record in the resolved_df